src/Version.h

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (C) 2021 German Aerospace Center (DLR/SC)
+*
+* Created: 2021-11-02 Martin Siggel <Martin.Siggel@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef VERSION_H
+#define VERSION_H
+
+#include <string>
+#include <regex>
+#include <exception>
+
+/**
+ * @brief A class to parse and compare version strings
+ */
+class Version
+{
+public:
+    explicit Version(const std::string& str)
+        : m_major(0),
+          m_minor(0),
+          m_patch(0)
+    {
+        parse(str);
+    }
+
+    int vMajor() const
+    {
+        return m_major;
+    }
+
+    int vMinor() const
+    {
+        return m_minor;
+    }
+
+    int vPatch() const
+    {
+        return m_patch;
+    }
+
+    std::string vLabel() const
+    {
+        return m_label;
+    }
+
+    std::string vBuild() const
+    {
+        return m_build;
+    }
+
+    bool operator< (const Version& other) const
+    {
+        if (vMajor() < other.vMajor()) {
+            return true;
+        }
+        else if (vMajor() > other.vMajor()) {
+            return false;
+        }
+
+        // major Versions are same
+        if (vMinor() < other.vMinor()) {
+            return true;
+        }
+        else if (vMinor() > other.vMinor()) {
+            return false;
+        }
+
+        // major and minor are same
+        if (vPatch() < other.vPatch()) {
+            return true;
+        }
+        else if (vPatch() > other.vPatch()) {
+            return false;
+        }
+
+        // we don't compare build or pre-release right now
+        // is there any sane way to do it???
+
+
+        return false;
+
+    }
+
+    bool operator> (const Version& other) const
+    {
+        return other < *this;
+    }
+
+    bool operator== (const Version& other) const
+    {
+        return !(*this < other) && !(*this > other);
+    }
+
+private:
+
+    int m_major, m_minor, m_patch;
+    std::string m_label, m_build;
+
+    void parse(const std::string& str)
+    {
+        const auto expr = std::regex("([0-9]+)\\.([0-9]+)(\\.([0-9]+))?(?:-(\\w+(\\.\\w*)?)(\\+(.*))?|\\+(.*))?");
```

Comment:
I wonder how strict we want to enforce semver. semver.org provides a different regex: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string and some examples of valid versions and invalid versions here: https://regex101.com/r/vkijKf/1/

Here is how the regex used here performs on the semver-provided examples: https://regex101.com/r/XjshXb/1
There are some discrepancies:

 - The regex here accepts versions without a patch, while the semver regex apparently does not. On the one hand, I think it is reasonable to accept MAJOR.MINOR version strings. On the other hand, we would have to define precende rules for this case. In the current implementation, I assume MAJOR.MINOR is treated like MAJOR.MINOR.0, right? So "3.2" is always less than "3.2.x" if x>0?
 - There are small details, such as the pre-release label may not begin with zero, or some symbols are not allowed here or there.

Reply:
> I wonder how strict we want to enforce semver. semver.org provides a different regex: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string and some examples of valid versions and invalid versions here: https://regex101.com/r/vkijKf/1/

We need to support versions like "3.2" which is no valid semver i think, is it? Hence, i already use a modified version.
Despite that, we could copy the "official" semver regex and modify it to allow Major.Minor Versions as well. What do you think?

Reply:
> copy the "official" semver regex and modify it to allow Major.Minor Versions as well

I agree, that sounds like the cleanest solution. Should we document this in the docstring of the version class? E.g. *"this class implements semver 2.0.0 with the following modifications: ..."*

Reply:
Yes!